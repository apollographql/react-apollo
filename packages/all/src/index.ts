// Common + hooks
export * from '@apollo/react-common';

// @apollo/react-components
export {
  Query,
  Mutation,
  Subscription,
  QueryComponentOptions,
  MutationComponentOptions,
  SubscriptionComponentOptions
} from '@apollo/react-components';

// @apollo/react-hoc
export {
  graphql,
  withQuery,
  withMutation,
  withSubscription,
  withApollo,
  QueryControls,
  DataValue,
  DataProps,
  MutateProps,
  ChildProps,
  ChildDataProps,
  ChildMutateProps,
  OptionProps,
  OperationOption,
  WithApolloClient
} from '@apollo/react-hoc';

// SSR
export {
  getMarkupFromTree,
  getDataFromTree,
  renderToStringWithData
} from '@apollo/react-ssr';
