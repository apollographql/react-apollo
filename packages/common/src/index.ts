export {
  ApolloContextValue,
  ApolloContextOptions,
  getApolloContext,
  resetApolloContext
} from './context/ApolloContext';
export { ApolloProvider } from './context/ApolloProvider';
export { ApolloConsumer } from './context/ApolloConsumer';

export {
  parser,
  operationName,
  DocumentType,
  IDocumentDefinition
} from './parser/parser';

export * from './types/types';
