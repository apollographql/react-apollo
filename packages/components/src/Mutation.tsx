import { OperationVariables } from '@apollo/react-common';
import { useMutation } from '@apollo/react-hooks';
import PropTypes from 'prop-types';

import { MutationComponentOptions } from './types';

export function Mutation<TData = any, TVariables = OperationVariables>(
  props: MutationComponentOptions<TData, TVariables>
) {
  const [runMutation, result] = useMutation(props.mutation, props);
  return props.children ? props.children(runMutation, result) : null;
}

export namespace Mutation {
  export const propTypes = {
    mutation: PropTypes.object.isRequired,
    variables: PropTypes.object,
    optimisticResponse: PropTypes.object,
    refetchQueries: PropTypes.oneOfType([
      PropTypes.arrayOf(
        PropTypes.oneOfType([PropTypes.string, PropTypes.object])
      ),
      PropTypes.func
    ]),
    awaitRefetchQueries: PropTypes.bool,
    update: PropTypes.func,
    children: PropTypes.func.isRequired,
    onCompleted: PropTypes.func,
    onError: PropTypes.func,
    fetchPolicy: PropTypes.string
  };
}
