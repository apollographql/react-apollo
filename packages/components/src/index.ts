export {
  ApolloProvider,
  ApolloConsumer,
  getApolloContext,
  resetApolloContext
} from '@apollo/react-common';

export { Query } from './Query';
export { Mutation } from './Mutation';
export { Subscription } from './Subscription';

export * from './types';
