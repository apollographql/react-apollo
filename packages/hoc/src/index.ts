export { graphql } from './graphql';

export { withQuery } from './query-hoc';
export { withMutation } from './mutation-hoc';
export { withSubscription } from './subscription-hoc';
export { withApollo } from './withApollo';

export {
  ApolloProvider,
  ApolloConsumer,
  getApolloContext,
  resetApolloContext
} from '@apollo/react-common';

export * from './types';
