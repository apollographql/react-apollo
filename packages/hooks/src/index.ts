export {
  ApolloProvider,
  ApolloConsumer,
  getApolloContext,
  resetApolloContext
} from '@apollo/react-common';

export { useQuery } from './useQuery';
export { useLazyQuery } from './useLazyQuery';
export { useMutation } from './useMutation';
export { useSubscription } from './useSubscription';
export { useApolloClient } from './useApolloClient';

export { RenderPromises } from './ssr/RenderPromises';

export * from './types';
