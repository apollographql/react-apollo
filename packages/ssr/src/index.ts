export { getMarkupFromTree, getDataFromTree } from './getDataFromTree';
export { renderToStringWithData } from './renderToStringWithData';
export * from './types';
