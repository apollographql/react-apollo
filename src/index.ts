export * from './browser';
export * from './server';
