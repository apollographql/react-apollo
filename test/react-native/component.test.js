import 'react-native';
import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View
} from 'react-native';
// import { mount, shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import ApolloClient from 'apollo-client';
import gql from 'graphql-tag';
import { ApolloProvider, graphql } from '../../lib/src';
import mockNetworkInterface from '../../lib/test/mocks/mockNetworkInterface';

describe('App', () => {

  it('renders correctly', () => {
    const query = gql`query people { allPeople(first: 1) { people { name } } }`;
    const data = { allPeople: { people: { name: 'Luke Skywalker' } } };
    const networkInterface = mockNetworkInterface({ request: { query }, result: { data } });
    const client = new ApolloClient({ networkInterface });

    const ContainerWithData = graphql(query)(({ data }) => {
      if (data.loading) return <Text>Loading...</Text>;
      return <Text>{data.allPeople.people.name}</Text>;
    });
    const output = renderer.create(
      <ApolloProvider client={client}><ContainerWithData /></ApolloProvider>
    )
    expect(output.toJSON()).toMatchSnapshot();
  });

  it('executes a query', (done) => {
    jest.useRealTimers()
    const query = gql`query people { allPeople(first: 1) { people { name } } }`;
    const data = { allPeople: { people: { name: 'Luke Skywalker' } } };
    const networkInterface = mockNetworkInterface({ request: { query }, result: { data } });
    const client = new ApolloClient({ networkInterface });

    class Container extends Component {
      componentWillReceiveProps(props) {
        expect(props.data.loading).toBeFalsy();
        expect(props.data.allPeople.people.name).toEqual(data.allPeople.people.name);
        done();
      }
      render() {
        if (this.props.data.loading) return <Text>Loading...</Text>;
        return <Text>{this.props.data.allPeople.people.name}</Text>;
      }
    };

    const ContainerWithData = graphql(query)(Container);

    const output = renderer.create(
      <ApolloProvider client={client}><ContainerWithData /></ApolloProvider>
    )
  });

});
