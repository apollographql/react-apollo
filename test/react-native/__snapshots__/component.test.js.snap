exports[`App renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  lineBreakMode="tail">
  Loading...
</Text>
`;
